apps/builder/app/builder/features/sidebar-left/panels/navigator/navigator.tsx

```diff
@@ -1,12 +1,22 @@
 import { ListNestedIcon } from "@webstudio-is/icons";
-import { Navigator } from "../../navigator";
+import { Navigator, useNavigatorLayout } from "../../navigator";
 import type { TabName } from "../../types";
+import { useEffect } from "react";
 
 type TabContentProps = {
   onSetActiveTab: (tabName: TabName) => void;
 };
 
 export const TabContent = ({ onSetActiveTab }: TabContentProps) => {
+  const navigatorLayout = useNavigatorLayout();
+
+  useEffect(() => {
+    // When user switches to undocked navigator mode, we need to close the tab.
+    if (navigatorLayout === "undocked") {
+      onSetActiveTab("none");
```

Comment:
Since it's possible to break this by passing arrow which is created on every render I would suggest to call store.set here directly.